sql/src/main/java/io/crate/planner/consumer/ManyTableConsumer.java

```diff
@@ -250,6 +250,7 @@ static TwoTableJoin buildTwoTableJoinTree(MultiSourceSelect mss) {
             currentTreeRelationNames.add(rightName);
 
             // process where clause
+            pushDownNoMatch(mss, leftRelation, rightRelation);
```

Comment:
hm, it's not the same.
We still need to push down the no_match to the rightrelation so that the collectphase created out of it returns no rows.